### DISCUSSION TITLE
Mirth Client API: NullPointer when searching messages

### DISCUSSION THREAD

**Discussion by JoeFox82** [nextgenhealthcare/connect#5117] [Q&A] [answered]
Hi together,

when trying the _​/channels​/{channelId}​/messages_ function I always get a NullPointerException with error code 500 Internal Server Error. I'm logged in as admin and all other functions (e.g. /channels/{channelId}) are working well. The channel with the channelId is running and when I use the search function in the dashboard using the same search parameters as is used in the API I get the messages displayed.

I'm using version 3.12

Request url:
https://localhost:8443/api/channels/72334f65-30a4-4be6-9ef8-46663ee1e9d6/messages?rawContentSearch=151333978&limit=20

`
<java.lang.NullPointerException>
  <stackTrace>
    <trace>com.mirth.connect.server.controllers.DonkeyMessageController.getMessages(DonkeyMessageController.java:197)</trace>
    <trace>com.mirth.connect.server.api.servlets.MessageServlet.getMessages(MessageServlet.java:169)</trace>
    <trace>sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)</trace>
    <trace>sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)</trace>
    <trace>sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)</trace>
    <trace>java.lang.reflect.Method.invoke(Method.java:498)</trace>
    <trace>com.mirth.connect.server.api.providers.MirthResourceInvocationHandlerProvider$1.invoke(MirthResourceInvocationHandlerProvider.java:219)</trace>
    <trace>org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)</trace>
    <trace>org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)</trace>
    <trace>org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205)</trace>
    <trace>org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)</trace>
    <trace>org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)</trace>
    <trace>org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)</trace>
    <trace>org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)</trace>
    <trace>org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326)</trace>
    <trace>org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)</trace>
    <trace>org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)</trace>
    <trace>org.glassfish.jersey.internal.Errors.process(Errors.java:315)</trace>
    <trace>org.glassfish.jersey.internal.Errors.process(Errors.java:297)</trace>
    <trace>org.glassfish.jersey.internal.Errors.process(Errors.java:267)</trace>
    <trace>org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)</trace>
    <trace>org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)</trace>
    <trace>org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)</trace>
    <trace>org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:471)</trace>
    <trace>org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:425)</trace>
    <trace>org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:383)</trace>
    <trace>org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:336)</trace>
    <trace>org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:223)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:760)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1617)</trace>
    <trace>com.mirth.connect.server.api.providers.StrictTransportSecurityFilter.doFilter(StrictTransportSecurityFilter.java:33)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)</trace>
    <trace>com.mirth.connect.server.MethodFilter.doFilter(MethodFilter.java:37)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)</trace>
    <trace>com.mirth.connect.server.api.providers.RequestedWithFilter.doFilter(RequestedWithFilter.java:53)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)</trace>
    <trace>com.mirth.connect.server.api.providers.ClickjackingFilter.doFilter(ClickjackingFilter.java:44)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)</trace>
    <trace>com.mirth.connect.server.api.providers.ApiOriginFilter.doFilter(ApiOriginFilter.java:71)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)</trace>
    <trace>org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)</trace>
    <trace>org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1589)</trace>
    <trace>org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)</trace>
    <trace>org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1296)</trace>
    <trace>org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)</trace>
    <trace>org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)</trace>
    <trace>org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1559)</trace>
    <trace>org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)</trace>
    <trace>org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1211)</trace>
    <trace>org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)</trace>
    <trace>org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:59)</trace>
    <trace>org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)</trace>
    <trace>org.eclipse.jetty.server.Server.handle(Server.java:500)</trace>
    <trace>org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:386)</trace>
    <trace>org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:560)</trace>
    <trace>org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:378)</trace>
    <trace>org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268)</trace>
    <trace>org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)</trace>
    <trace>org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)</trace>
    <trace>org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:426)</trace>
    <trace>org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:320)</trace>
    <trace>org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:158)</trace>
    <trace>org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)</trace>
    <trace>org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)</trace>
    <trace>org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)</trace>
    <trace>org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)</trace>
    <trace>org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)</trace>
    <trace>org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)</trace>
    <trace>org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367)</trace>
    <trace>org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782)</trace>
    <trace>org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:914)</trace>
    <trace>java.lang.Thread.run(Thread.java:750)</trace>
  </stackTrace>
  <suppressedExceptions class="java.util.Collections$UnmodifiableRandomAccessList" resolves-to="java.util.Collections$UnmodifiableList">
    <c class="list"/>
    <list/>
  </suppressedExceptions>
</java.lang.NullPointerException>
`

Any ideas?

Kind regards,
Jochen

**Comment by jonbartels** [comment, marked as answer]
Are you having the same problem as https://github.com/nextgenhealthcare/connect/discussions/4638 ?

**Reply by JoeFox82** [reply]
Thank you very much, I did not find that discussion by searching all discussions!
